pkg/storage/fs/cback/utilities.go

```diff
@@ -0,0 +1,360 @@
+	case "file":
+		return provider.ResourceType_RESOURCE_TYPE_FILE, nil
+
+	case "dir":
+		return provider.ResourceType_RESOURCE_TYPE_CONTAINER, nil
+
+	default:
+		return provider.ResourceType_RESOURCE_TYPE_INVALID, errtypes.NotFound("Resource type unrecognized")
+	}
+}
+
+func (fs *cback) getRequest(userName, url string, reqType string, body io.Reader) (io.ReadCloser, error) {
+
+	req, err := http.NewRequest(reqType, url, body)
+	req.SetBasicAuth(userName, fs.conf.ImpersonatorToken)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if body != nil {
+		req.Header.Add("Content-Type", "application/json")
+	}
+
+	req.Header.Add("accept", `application/json`)
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == 404 {
+		return nil, errtypes.NotFound("cback: resource not found")
+	}
+
+	if resp.StatusCode == 500 {
+		return nil, errtypes.InternalError("cback: internal server error")
+	}
+
+	if resp.StatusCode == 403 {
+		return nil, errtypes.PermissionDenied("cback: user has no permissions to get the backup")
+	}
+
+	if resp.StatusCode == 400 {
+		return nil, errtypes.BadRequest("cback")
+	}
+
+	return resp.Body, nil
+}
+
+func (fs *cback) listSnapshots(userName string, backupID int) ([]snapshotResponse, error) {
+
+	url := fs.conf.APIURL + "/backups/" + strconv.Itoa(backupID) + "/snapshots"
+	requestType := "GET"
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []snapshotResponse{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	return responseObject, nil
+}
+
+func (fs *cback) matchBackups(userName, pathInput string) (*backUpResponse, error) {
+
+	url := fs.conf.APIURL + "/backups/"
+	requestType := "GET"
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []backUpResponse{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	if len(responseObject) == 0 {
+		err = errors.New("no match found")
+		return nil, err
+	}
+
+	for i := range responseObject {
+		if responseObject[i].Detail != "" {
+			err = errors.New(responseObject[i].Detail)
+			return nil, err
+		}
+
+		if strings.Compare(pathInput, responseObject[i].Source) == 0 {
+			return &responseObject[i], nil
+		}
+	}
+
+	for i := range responseObject {
+		if responseObject[i].Detail != "" {
+			err = errors.New(responseObject[i].Detail)
+			return nil, err
+		}
+
+		if strings.HasPrefix(pathInput, responseObject[i].Source) {
+			substr := strings.TrimPrefix(pathInput, responseObject[i].Source)
+			substr = strings.TrimLeft(substr, "/")
+			responseObject[i].Substring = substr
+			return &responseObject[i], nil
+		}
+	}
+
+	return nil, nil
```

Comment:
please document what means a nil response

Reply:
`	/*If there is no error, but also no match found in the backup path the response is nil.
	This means that the LSFolder function will know that no match has been found using the exact path,
	and will therefore start checking if there is a substring of the backup job included in the inputted path*/`